### PR TITLE
chore(datadog): Move fetch functions, add back codegen for services

### PR DESCRIPTION
**Pull request by hermanschaaf**
This change:
 - restores the services codegen code accidentally removed in https://github.com/cloudquery/cloudquery/pull/6424. We are still relying on it for the generation of service interfaces and mocks
 - moves resolvers out of `_fetch.go` files into the main resource files, as is now standard practice
 - updates the contribution guide

There are no user-facing changes here, only refactoring and developer-facing changes.

### CODE REVIEW

File: plugins/source/datadog/codegen/services/generate.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
+package services
```
Comment (hermanschaaf) [inline plugins/source/datadog/codegen/services/generate.go:1]: This file was just restored from the Git history, no changes

File: plugins/source/datadog/CONTRIBUTING.md [hunk 2]
```diff
@@ -2,100 +2,64 @@
...
+ - A working Go installation (1.19+). See [here](https://go.dev/doc/install) for instructions.
 
+## Adding a new resource
 
-## 1. Generate Interfaces for the Datadog SDK Function(s) that Fetch the Resource
+### Step 1. Add the service to the client
 
-### Before you Start
+When adding support for new APIs, you may need to add a new service to the client. To do this:
+ 1. open `codegen/services/clients.go` and add the client from the Datadog SDK to the list of clients
```
Comment (candiduslynx) [inline plugins/source/datadog/CONTRIBUTING.md:14]:
```suggestion
 1. Open `codegen/services/clients.go` and add the client from the Datadog SDK to the list of clients
```

File: plugins/source/datadog/CONTRIBUTING.md [hunk 3]
```diff
@@ -2,100 +2,64 @@
...
+ 2. run `make gen-mocks` to generate the service interface and mocks
```
Comment (candiduslynx) [inline plugins/source/datadog/CONTRIBUTING.md:15]:
```suggestion
 2. Run `make gen-mocks` to generate the service interface and mocks
```

File: plugins/source/datadog/go.mod [hunk 4]
```diff
@@ -5,6 +5,7 @@ go 1.19
 require (
 	github.com/DataDog/datadog-api-client-go/v2 v2.9.0
 	github.com/cloudquery/plugin-pb-go v1.0.8
+	github.com/cloudquery/plugin-sdk v1.24.0
```
Comment (candiduslynx) [inline plugins/source/datadog/go.mod:8]: I really think this shouldn't be here anymore

File: plugins/source/datadog/codegen/services/generate.go [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+	"fmt"
+	"go/format"
+	"os"
+	"path"
+	"reflect"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/cloudquery/plugin-sdk/caser"
```
Comment (candiduslynx) [inline plugins/source/datadog/codegen/services/generate.go:15]:
```suggestion
	"github.com/cloudquery/plugin-sdk/v2/caser"
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]
needs to point to plugin-sdk/v2, others are nits

File: plugins/source/datadog/CONTRIBUTING.md (hunk 3 above)
Comment (hermanschaaf) [reply plugins/source/datadog/CONTRIBUTING.md]: Updated, thanks for the suggestion

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**